src/Task/ConvertTranslatableTask.php

```diff
@@ -0,0 +1,268 @@
+            throw new Exception(
+                "Please configure a Fluent default locale (in the CMS) prior to migrating from translatable"
+            );
+        }
+    }
+
+    /**
+     * Do something inside a DB transaction
+     *
+     * @param callable $callback
+     * @throws \Exception
+     */
+    protected function withTransaction($callback)
+    {
+        try {
+            Debug::message('Beginning transaction', false);
+            DB::get_conn()->transactionStart();
+            $callback($this);
+            Debug::message('Comitting transaction', false);
+            DB::get_conn()->transactionEnd();
+        } catch (\Exception $ex) {
+            Debug::message('Rolling back transaction', false);
+            DB::get_conn()->transactionRollback();
+            throw $ex;
+        }
+    }
+
+    protected $translatedFields = [];
+
+    /**
+     * Get all database fields to translate
+     *
+     * @param string $class Class name
+     * @return array List of translated fields
+     */
+    public function getTranslatedFields($class)
+    {
+        if (isset($this->localisedFields[$class])) {
+            return $this->localisedFields[$class];
+        }
+        $fields = [];
+        $hierarchy = ClassInfo::ancestry($class);
+        foreach ($hierarchy as $class) {
+            // Skip classes without tables
+            if (!DataObject::getSchema()->classHasTable($class)) {
+                continue;
+            }
+
+            // Check fields localised by Fluent for this class
+            $localisedFields = singleton($class)->getLocalisedFields();
+            if (empty($localisedFields)) {
+                continue;
+            }
+
+            // Save fields
+            $fields = array_merge($fields, array_keys($localisedFields));
+        }
+        $this->localisedFields[$class] = $fields;
+        return $fields;
+    }
+
+    /**
+     * Gets all classes with FluentExtension
+     *
+     * @return array Array of classes to migrate
+     */
+    public function fluentClasses()
+    {
+        $classes = [];
+        $dataClasses = ClassInfo::subclassesFor(DataObject::class);
+        array_shift($dataClasses);
+        foreach ($dataClasses as $class) {
+            $base = DataObject::getSchema()->baseDataClass($class);
+            foreach (DataObject::get_extensions($base) as $extension) {
+                if (is_a($extension, FluentExtension::class, true)) {
+                    $classes[] = $base;
+                    break;
+                }
+            }
+        }
+        return array_unique($classes);
+    }
+
+    public function run($request)
+    {
+        // Extend time limit
+        set_time_limit(100000);
+
+        // we may need some privileges for this to work
+        // without this, running under sake is a problem
+        // maybe sake could take care of it ...
+        Security::setCurrentUser(
+            DefaultAdminService::singleton()->findOrCreateDefaultAdmin()
```

Comment:
Will fail if default admin isn't enabled. Will need a work-around.

Reply:
We could just push a message and return early in this case

Reply:
findOrCreateAdmin() works as a failover ;P

Reply:
Lost you - leave it as it?